relayer/pkg/chainlink/txm/txm.go

```diff
@@ -204,21 +205,20 @@ func (txm *starktxm) confirmLoop() {
 			for addr := range hashes {
 				for i := range hashes[addr] {
 					hash := hashes[addr][i]
-					status, err := client.Gw.TransactionStatus(ctx, gateway.TransactionStatusOptions{
-						TransactionHash: hashes[addr][i],
-					})
+					response, err := client.Provider.TransactionReceipt(ctx, caigotypes.StrToFelt(hashes[addr][i]))
 					if err != nil {
 						txm.lggr.Errorw("failed to fetch transaction status", "hash", hash, "error", err)
 						continue
 					}
-
-					if status == nil {
-						txm.lggr.Errorw("status was nil", "hash", hash)
-						continue
+					receipt, ok := response.(caigorpc.InvokeTransactionReceipt)
+					if !ok {
+						panic(fmt.Sprintf("wrong receipt type: %v", reflect.TypeOf(response)))
 					}
 
-					if status.TxStatus == "ACCEPTED_ON_L1" || status.TxStatus == "ACCEPTED_ON_L2" || status.TxStatus == "REJECTED" {
-						txm.lggr.Debugw(fmt.Sprintf("tx confirmed: %s", status.TxStatus), "hash", hash, "status", status)
+					status := receipt.Status
+
+					if status == caigotypes.TransactionAcceptedOnL1 || status == caigotypes.TransactionAcceptedOnL2 || status == caigotypes.TransactionRejected {
```

Comment:
@aalu1418 it seems weird to me that rejected is just treated as confirmed?

Reply:
rejection is still technically an inclusion of the tx in the blockchain - if i understood correctly, rejection == reverts